Kernel/Graphics/GraphicsManagement.cpp

```diff
@@ -56,7 +56,74 @@ void GraphicsManagement::activate_graphical_mode()
 
 static inline bool is_vga_compatible_pci_device(PCI::Address address)
 {
-    return PCI::get_class(address) == 0x3 && PCI::get_subclass(address) == 0x0;
+    // Note: Check for Display Controller, VGA Compatible Controller or
+    // Unclassified, VGA-Compatible Unclassified Device
+    return (PCI::get_class(address) == 0x3 && PCI::get_subclass(address) == 0x0) || (PCI::get_class(address) == 0x0 && PCI::get_subclass(address) == 0x1);
+}
+
+static inline bool is_display_controller_pci_device(PCI::Address address)
+{
+    return PCI::get_class(address) == 0x3;
+}
+
+UNMAP_AFTER_INIT bool GraphicsManagement::determine_and_initialize_graphics_device(const PCI::Address& address, PCI::ID id)
+{
+    VERIFY(is_vga_compatible_pci_device(address) || is_display_controller_pci_device(address));
+    auto add_and_configure_adapter = [&](GraphicsDevice& graphics_device) {
+        m_graphics_devices.append(graphics_device);
+        if (!m_framebuffer_devices_allowed) {
+            graphics_device.enable_consoles();
+            return;
+        }
+        graphics_device.initialize_framebuffer_devices();
+    };
+
+    RefPtr<GraphicsDevice> adapter;
+    switch (id.vendor_id) {
+    case 0x1234:
+        if (id.device_id == 0x1111)
+            adapter = BochsGraphicsAdapter::initialize(address);
+        break;
+    case 0x80ee:
+        if (id.device_id == 0xbeef)
+            adapter = BochsGraphicsAdapter::initialize(address);
+        break;
+    case 0x8086:
+        adapter = IntelNativeGraphicsAdapter::initialize(address);
+        break;
+    case static_cast<u16>(PCIVendorID::VirtIO):
```

Comment:
Perhaps we could make `PCIVendorID` a `u16` (as in `enum class PCIVendorID : u16`) and use `to_underlying(PCIVendorID::VirtIO)` here instead of an explicit cast?